features/support/api_helpers.rb

```diff
@@ -120,6 +120,37 @@ def submit_supplier_declaration(framework_slug, supplier_id, declaration)
   })
   [200, 201].should include(response.code), _error(response, "Failed to submit supplier declaration #{framework_slug} #{supplier_id}")
   JSON.parse(response.body)['declaration']
+  set_supplier_on_framework(framework_slug, supplier_id, true)
```

Comment:
I don't think setting `on_framework=True` should be part of `submit_supplier_declaration` - you can submit a declaration and then not be awarded onto a framework.

I think it makes more sense to the `that supplier is on that framework` step - it could submit the declaration (as it already does) and then have an additional call to `set_supplier_on_framework`.